src/gui/FileBrowser.cpp

```diff
@@ -326,6 +328,49 @@ FileBrowserTreeWidget::FileBrowserTreeWidget(QWidget * parent ) :
 
 }
 
+QList<QString> FileBrowserTreeWidget::expandedDirs() const
+{
+	QList<QString> expandedDirs;
+	for( int i = 0; i < topLevelItemCount(); ++i )
+	{
+		QTreeWidgetItem * item  = topLevelItem( i );
+
+		// Add expanded top level directories.
+		if (item->isExpanded() && (item->type() == TypeDirectoryItem))
+		{
+			Directory *d = static_cast<Directory *> ( item );
```

Comment:
Yes it is intended. Since the check `if (item->isExpanded() && (item->type() == TypeDirectoryItem))` we can be very certain that it’s type is `Directory`, so I thought no need to do a `dynamic_cast`. Please do correct me if I’m wrong.